OpenRA.Mods.Common/Widgets/Logic/Editor/SaveMapLogic.cs

```diff
@@ -239,41 +236,119 @@ public SaveDirectory(Folder folder, string displayName, MapClassification classi
+						{
+							saveMap(combinedPath);
+							if (actionManager != null)
+								actionManager.SaveFailed = false;
+						},
+						onCancel: () =>
+						{
+							if (actionManager != null)
+								actionManager.SaveFailed = false;
+						});
+
+					if (actionManager != null)
+						actionManager.SaveFailed = true;
+
+					return;
 				}
-				else
+			}
+			else
+			{
+				// When file paths match
+				var recentUid = modData.MapCache.GetUpdatedMap(map.Uid);
+				if (recentUid != null && map.Uid != recentUid && modData.MapCache[recentUid].Status == MapStatus.Available)
 				{
-					// When file paths match
-					var recentUid = modData.MapCache.GetUpdatedMap(map.Uid);
-					if (recentUid != null && map.Uid != recentUid && modData.MapCache[recentUid].Status == MapStatus.Available)
-					{
-						ConfirmationDialogs.ButtonPrompt(modData,
-							title: OverwriteMapOutsideEditTitle,
-							text: OverwriteMapOutsideEditPrompt,
-							confirmText: SaveMapMapOutsideConfirm,
-							onConfirm: () => saveMap(combinedPath),
-							onCancel: () => { });
-
-						return;
-					}
+					ConfirmationDialogs.ButtonPrompt(modData,
+						title: OverwriteMapOutsideEditTitle,
+						text: OverwriteMapOutsideEditPrompt,
+						confirmText: SaveMapMapOutsideConfirm,
+						onConfirm: () =>
+						{
+							saveMap(combinedPath);
+							if (actionManager != null)
+								actionManager.SaveFailed = false;
+						},
+						onCancel: () =>
+						{
+							if (actionManager != null)
+								actionManager.SaveFailed = false;
+						});
+
+					if (actionManager != null)
+						actionManager.SaveFailed = true;
+
+					return;
 				}
+			}
 
-				saveMap(combinedPath);
-			};
+			saveMap(combinedPath);
+		}
+
+		public static void SaveMapInner(Map map, IReadWritePackage package, World world, ModData modData)
+		{
+			map.RequiresMod = modData.Manifest.Id;
+
+			try
+			{
+				if (package == null)
+					throw new ArgumentNullException(nameof(package));
+
+				map.Save(package);
+
+				var actionManager = world.WorldActor.TraitOrDefault<EditorActionManager>();
+				if (actionManager != null)
+					actionManager.Modified = false;
+
+				TextNotificationsManager.AddTransientLine(modData.Translation.GetString(SaveCurrentMap), world.LocalPlayer);
+			}
+			catch (Exception e)
+			{
+				SaveMapFailed(e, modData, world);
+			}
+		}
+
+		static void SaveMapFailed(Exception e, ModData modData, World world)
+		{
+			Log.Write("debug", $"Failed to save map.");
+			Log.Write("debug", e);
+
+			var actionManager = world.WorldActor.TraitOrDefault<EditorActionManager>();
+			if (actionManager != null)
+				actionManager.SaveFailed = true;
+
+			ConfirmationDialogs.ButtonPrompt(modData,
+				title: SaveMapFailedTitle,
+				text: SaveMapFailedPrompt,
+				onConfirm: () =>
+				{
+					if (actionManager != null)
+						actionManager.SaveFailed = false;
+				},
+				onCancel: () =>
+				{
+					if (actionManager != null)
+						actionManager.SaveFailed = false;
+				},
```

Comment:
I don't think we need the double button prompt here. There's nothing to cancel when a map fails to save.
![image](https://user-images.githubusercontent.com/1355810/207009479-db6206f2-84e4-4a41-88f6-7388acbccfa7.png)